spec/compiler/codegen/struct_spec.cr

```diff
@@ -582,4 +582,28 @@ describe "Code gen: struct" do
       f.x
       )).to_i.should eq(84)
   end
+
+  it "can cast virtual struct to specific struct" do
```

Comment:
I thought this broke with this PR but it seems it was a general bug. I fixed it in master now, thanks! :-)